xcresult/main.swift

```diff
@@ -8,9 +8,31 @@
 
 import Foundation
 
+enum Flag: String {
+	case machineFriendly = "-m"
+	case `default` = ""
+	
+	init(rawValue: String) {
+		switch rawValue {
+		case "-m": self = .machineFriendly
+		default: self = .default
+		}
+	}
+}
+
 if let resultArgument = ProcessInfo.processInfo.arguments[safeAt: 1] {
-    let url = URL(fileURLWithPath: resultArgument)
-    ResultHandler().handle(resultURL: url)
+	let result = Flag(rawValue: resultArgument)
+	let shouldOpenResultFile = result == .default
+	switch result {
+	case .machineFriendly:
+		if let urlString = ProcessInfo.processInfo.arguments[safeAt: 2] {
+			let url = URL(fileURLWithPath: urlString)
+			ResultHandler().handle(resultURL: url, shouldOpenResultFile: shouldOpenResultFile)
+		}
+	default:
```

Comment:
Could we use `case .default` instead here? That way if a new case gets added the compiler will enforce this to be thought about